aten/src/ATen/native/DistributionTemplates.h

```diff
@@ -110,13 +111,20 @@ static void check_from_to_in_range(int64_t from, int64_t to_inc, caffe2::TypeMet
       WARN_OUT_OF_BOUNDS(from, "from", digits, dtype);
       WARN_OUT_OF_BOUNDS(to_inc, "to - 1", digits, dtype);
     });
+  } else if (scalar_type == kUInt64) {
+    AT_DISPATCH_V2(scalar_type, "check_random_integral_bounds", AT_WRAP([&]() {
+      // This is annoying: because randint takes int arguments, this means
```

Comment:
Why can't you use UINT64_T bounds here?
the min will be 0 and the max will be larger than INT64_MAX so it is ok to use.
Or it least to bad implicit casting of "from" and "to_inc" that are problematic?

Reply:
This piece is sooooo annoying.

When you do a comparison between int64_t and uint64_t, the usual arithmetic conversions say that the int64_t value is promoted to unsigned. But this conversion wraps around: if I had -1 as my int64_t, then it will promote to 0xFFFFFFFFFFFFFFFF in uint64_t. This is never the right thing to do.